repotracker/repotracker.go

```diff
@@ -717,123 +719,144 @@ func createVersionItems(ctx context.Context, v *model.Version, ref *model.Projec
 	if err != nil {
 		return errors.WithStack(err)
 	}
+	// generate all task Ids so that we can easily reference them for dependencies
+	sourceRev := ""
+	if metadata.SourceVersion != nil {
+		sourceRev = metadata.SourceVersion.Revision
+	}
+	taskIds := model.NewTaskIdTable(project, v, sourceRev, metadata.TriggerDefinitionID)
 
-	txFunc := func(sessCtx mongo.SessionContext) (bool, error) {
-		// generate all task Ids so that we can easily reference them for dependencies
-		sourceRev := ""
-		if metadata.SourceVersion != nil {
-			sourceRev = metadata.SourceVersion.Revision
-		}
-		taskIds := model.NewTaskIdTable(project, v, sourceRev, metadata.TriggerDefinitionID)
-
-		err := sessCtx.StartTransaction()
-		if err != nil {
-			return false, errors.Wrap(err, "error starting transaction")
+	// create all builds for the version
+	buildsToCreate := []interface{}{}
+	tasksToCreate := task.Tasks{}
+	for _, buildvariant := range project.BuildVariants {
```

Comment:
maybe check here if the context is canceled and abort earlier.

Reply:
added